File: matrix_addition.py
"""Compute addition of two matrices."""

matrix_a = [
    [1, 2, 3],
    [4, 5, 6],
    [7, 8, 9],
]
matrix_b = [
    [9, 8, 7],
    [6, 5, 4],
    [3, 2, 1],
]

if len(matrix_a) != len(matrix_b) or any(len(row_a) != len(row_b) for row_a, row_b in zip(matrix_a, matrix_b)):
    raise ValueError("Matrices must have the same dimensions")

result = [[row_a[j] + row_b[j] for j in range(len(row_a))] for row_a, row_b in zip(matrix_a, matrix_b)]

print("Matrix A:")
for row in matrix_a:
    print(row)
print("\nMatrix B:")
for row in matrix_b:
    print(row)
print("\nMatrix A + Matrix B:")
for row in result:
    print(row)

File: matrix_addition.cpp
#include <iostream>
#include <vector>

int main() {
    std::vector<std::vector<int>> matrixA = {
        {1, 2, 3},
        {4, 5, 6},
        {7, 8, 9}
    };
    std::vector<std::vector<int>> matrixB = {
        {9, 8, 7},
        {6, 5, 4},
        {3, 2, 1}
    };

    if (matrixA.size() != matrixB.size()) {
        std::cerr << "Matrices must have the same dimensions." << std::endl;
        return 1;
    }

    std::vector<std::vector<int>> result(matrixA.size());
    for (size_t i = 0; i < matrixA.size(); ++i) {
        if (matrixA[i].size() != matrixB[i].size()) {
            std::cerr << "Matrices must have the same dimensions." << std::endl;
            return 1;
        }
        result[i].resize(matrixA[i].size());
        for (size_t j = 0; j < matrixA[i].size(); ++j) {
            result[i][j] = matrixA[i][j] + matrixB[i][j];
        }
    }

    std::cout << "Matrix A:" << std::endl;
    for (auto &row : matrixA) {
        for (auto value : row) std::cout << value << " ";
        std::cout << std::endl;
    }

    std::cout << "\nMatrix B:" << std::endl;
    for (auto &row : matrixB) {
        for (auto value : row) std::cout << value << " ";
        std::cout << std::endl;
    }

    std::cout << "\nMatrix A + Matrix B:" << std::endl;
    for (auto &row : result) {
        for (auto value : row) std::cout << value << " ";
        std::cout << std::endl;
    }

    return 0;
}
